lib/integrations/awsoidc/deployservice_config.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsoidc
+
+import (
+	"encoding/base64"
+
+	"github.com/gravitational/trace"
+	"gopkg.in/yaml.v2"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/config"
+	"github.com/gravitational/teleport/lib/defaults"
+)
+
+// generateTeleportConfigString creates a teleport.yaml configuration
+func generateTeleportConfigString(req DeployServiceRequest) (string, error) {
+	teleportConfig, err := config.MakeSampleFileConfig(config.SampleFlags{
+		Version:      defaults.TeleportConfigVersionV3,
+		ProxyAddress: req.ProxyServerHostPort,
+	})
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	// Disable default services
+	teleportConfig.Auth.EnabledFlag = "no"
+	teleportConfig.Proxy.EnabledFlag = "no"
+	teleportConfig.SSH.EnabledFlag = "no"
+
+	// Ensure the NodeName is not set to the current host (Teleport Proxy).
+	// Setting it to an empty string, ensures the NodeName is picked up from the host's hostname.
+	teleportConfig.NodeName = ""
```

Comment:
Isn't it empty by default?

Reply:
It defaults to the current hostname, in this case is going to be the proxy's current hostname

https://github.com/gravitational/teleport/blob/e23e6b6082d133fa00dd191e123de56b579abb67/lib/config/fileconf.go#L213

If we don't override this to an empty string, it will keep that hostname as `nodename` in `teleport.yaml`
And, thus, every teleport instance will have the same nodename